### DISCUSSION TITLE
`SKXamlCanvas` for windows and wasm ???

### DISCUSSION THREAD

**Discussion by BertrandDeSaintBezier** [unoplatform/uno#13837] [Q&A]
Trying to load bitmaps on windows and wasm using the `SKSwapChainPanel` UI Element. In the shared project I've got the following to selectively include the right package based on the platform : 
``` xaml
<ItemGroup Condition="'$(TargetFramework)' == 'net7.0'">
	<PackageReference Include="Uno.WinUI.Runtime.WebAssembly" Version="4.9.26" />
	<PackageReference Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.6" />
</ItemGroup>
<ItemGroup Condition="'$(TargetFramework)' != 'net7.0'">
	<PackageReference Include="SkiaSharp.Views.WinUI" Version="2.88.6" />
</ItemGroup>
```
and then using it in code like such : 
```xaml
<Page ... 
    xmlns:skia="using:SkiaSharp.Views.Windows">
<skia:SKXamlCanvas x:Name="QRCode_Canvas" PaintSurface="SKXamlCanvas_PaintSurface" Width="500" Height="500"/>
``` 
This works fine on Windows, but I had a bunch of headaches concerning emscripten when trying to compile for wasm (had to install python, then node js and java, ended up compiling the emscripten source repo. Finally, Path variables had to be configured manually before I got the Uno project compiling correctly again). Now that the project is running on web again, it doesn't seem like the `SKXamlCanvas` is supported on web. I'm getting some error messages in the web console and nothing is showing up visually on my page + I can't get a reference to the Element from within `#if HAS_UNO` tags.

Not many up-to date resources on this, all I could find were mostly articles and discussions dating back to 2021, and almost exclusively using the old uno project structure with project heads. I found some mentions to `SKSWapChains` but this doesn't seem to exist in the `SkiaSharp.Views.Windows` namespace.

**Comment by jeromelaban** [comment]
Thanks for the questions. We're maintaining this sample here about SkiaSharp: https://github.com/unoplatform/Uno.Samples/tree/master/UI/SkiaSharpTest.

You should not be having to compile emscripten in any way, as the uno boostrapper is supposed to handle all of this automatically. Which errors do you get when building (Assuming you have python installed)?

SKSwapChain is not yet available on Windows at this time: https://github.com/mono/SkiaSharp/issues/1893

**Reply by BertrandDeSaintBezier** [reply]
At first I had the python error, and then I was getting an "unable to clone the emscripten sdk repo". I [found a post ](https://github.com/unoplatform/uno/discussions/11790) where you mentionned that deleting the .uno folder would fix the error. I tried this several times, but it didn't seem to do anything. Ultimately, I had to go through the process I outlined earlier of adding path variables etc. before I got it to compile correctly.

**Reply by jeromelaban** [reply]
Thanks for the update. I would be interesting to determine why you get this particular error message. If you can [provide a binlog](https://platform.uno/docs/articles/uno-builds-troubleshooting.html) of the error, it will be interesting.

**Reply by BertrandDeSaintBezier** [reply]
Dang, I'll have to break my project a little to try and reproduce this 😅 I'll get back to you on this as soon as I've managed to reproduce the error.
